packages/sanity/src/ui/menuItem/MenuItem.tsx

```diff
@@ -0,0 +1,161 @@
+import {
+  Flex,
+  MenuItem as UIMenuItem,
+  MenuItemProps as UIMenuItemProps,
+  Box,
+  Text,
+  Badge,
+  Stack,
+  Hotkeys,
+} from '@sanity/ui'
+import React, {createElement, forwardRef, isValidElement, useMemo} from 'react'
+import {isValidElementType} from 'react-is'
+
+interface LargeMenuItem {
+  size: 'large'
+  subText?: string
+  badgeText?: string
+  avatar?: React.ReactNode
+  /**
+   * Hotkeys are only supported in `size="small"` menu items.
+   */
+  hotkeys?: undefined
+  /**
+   * Icon is only supported in `size="small"` menu items.
+   */
+  icon?: undefined
+}
+
+interface SmallMenuItem {
+  size?: 'small'
+  hotkeys?: UIMenuItemProps['hotkeys']
+  /**
+   * Sub text is only supported in `size="large"` menu items.
+   */
+  subText?: undefined
+  /**
+   * Badge text is only supported in `size="large"` menu items.
+   */
+  badgeText?: undefined
+  /**
+   * Avatar is only supported in `size="large"` menu items.
+   */
+  avatar?: undefined
+}
+
+const fontSize = 1
```

Comment:
```suggestion
const FONT_SIZE = 1
```

**suggestion(non-blocking):** consider screaming snake case, which better indicates intent and is a pattern used throughout the studio